pygmt/src/meca.py

```diff
@@ -247,6 +240,29 @@ def update_pointers(data_pointers):
         ):
```

Comment:
Lines 237-241 are unnecessary because line 236 already checks `spec` is dict or pd.DataFrame. Am I right?

Reply:
You might be right, let me recheck the logic.

Reply:
@weiji14 Please see my new PR #1784.